### PR TITLE
32703 muon GUI to do async threads

### LINKED ISSUE #32703: Muon GUI to do async threads
The muon GUI should use the async threads (https://github.com/mantidproject/mantid/blob/main/qt/python/mantidqt/mantidqt/utils/async_qt_adaptor.py) and an example of  how to use it is https://github.com/mantidproject/mantid/blob/main/scripts/SANS/sans/gui_logic/models/async_workers/masking_table_async.py

### PR DESCRIPTION
**Pull request by Harrietbrown**
**Description of work.**

This PR changes the thredding used by the Muon GUI to use AsyncTaskQtAdaptor

**To test:**

1. build mantid in release and run through a profiler https://developer.mantidproject.org/ProfilingOverview.html
1. open muon analysis 
2. load MUSR62260-5
3. check profileing and see the run speed of functions coming from load_run_presenter.

Fixes #32703

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/python/mantidqtinterfaces/test/Muon/utilities/thread_model_test.py [hunk 1]
```diff
@@ -84,6 +76,13 @@ def mock_model(self):
...
 
+    def wait_for_thread(self, thread_model):
+        if thread_model and thread_model.worker:
+            while thread_model.worker.is_alive():
+                QApplication.sendPostedEvents()
+                time.sleep(0.1)
+            QApplication.sendPostedEvents()
+
     # ------------------------------------------------------------------------------------------------------------------
     # TESTS
```
Comment (sf1919) [inline qt/python/mantidqtinterfaces/test/Muon/utilities/thread_model_test.py:87]: Do we need to add any new tests now we're using async threading?

Comment (Harrietbrown) [reply qt/python/mantidqtinterfaces/test/Muon/utilities/thread_model_test.py:87]: With all the test that utalise the threads passing without having to change them beyond the wait_for_thread the existing tests should be sufficent.

**Comment by sf1919** [comment]
I have tested this PR using tests from recent Muon PRs (that have been mereged in prior to this work). They all worked as expected and certainly did not seem to be particularly slow. 

I then tested the PR using cProfile. The load_run_presenter functions in the profile all seem to be fairly quick - none of them struck me as being slow.

I've reviewed the code and it all looks good. Unnecessary code has been removed (specifically ```output```) and tests have been updated to take account of the new async threads. 

I am therefore happy to approve this PR which does represent an improvement in the speed of loading runs in the Mantid GUI but is also an improvement of the underlying code base. A good piece of work.

**Review by sf1919 — APPROVED** [review]
I have tested this PR using tests from recent Muon PRs (that have been mereged in prior to this work). They all worked as expected and certainly did not seem to be particularly slow.

I then tested the PR using cProfile. The load_run_presenter functions in the profile all seem to be fairly quick - none of them struck me as being slow.

I've reviewed the code and it all looks good. Unnecessary code has been removed (specifically output) and tests have been updated to take account of the new async threads.

I am therefore happy to approve this PR which does represent an improvement in the speed of loading runs in the Mantid GUI but is also an improvement of the underlying code base. A good piece of work.

**Comment by AnthonyLim23** [comment]
@mantidproject/gatekeepers could we please get this merged? It is holding up other work.

**Pull request merged**